File: java/contador.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        //Contador que llega hasta 30
        int contador = 0;
        while (contador < 30) {
            contador++;
            System.out.println("Iteración " + contador);
        }

        // Solicitar  valor mayor a 10
        Scanner scanner = new Scanner(System.in);
        int a;
        while (true) {
            System.out.print("Introduzca un valor mayor a 10: ");
            a = scanner.nextInt();

            if (a > 10) {
                System.out.println("Es correcto");
                break; // Sale del ciclo si la condición se cumple
            } else {
                System.out.println("Es incorrecto, pruebe de nuevo");
            }
        }

        // Cerrar el scanner
        scanner.close();
    }
}

File: cpp/contador.cpp
#include <iostream>
using namespace std;

int main()
{
    // Parte 1: Contador que llega hasta 30
    int contador = 0;
    while (contador < 30)
    {
        contador++;
        cout << "Iteración " << contador << endl;
    }

    // Parte 2: Solicitar al usuario un valor mayor a 10
    int a;
    while (true)
    {
        cout << "Introduzca un valor mayor a 10: ";
        cin >> a;

        if (a > 10)
        {
            cout << "Es correcto" << endl;
            break; // Sale del ciclo si la condición se cumple
        }
        else
        {
            cout << "Es incorrecto, pruebe de nuevo" << endl;
        }
    }

    return 0;
}